utils/generic.py

```diff
@@ -878,4 +932,152 @@ def assign_population_pcs(
     pop_pc_pd.loc[probs['max'] < min_prob, output_col] = missing_label
     pop_pc_pd = pop_pc_pd.drop(pc_cols, axis='columns')
 
-    return pop_pc_pd, pop_clf
+    logger.info("Found the following sample count after population assignment: {}".format(
+        ", ".join(f'{pop}: {count}' for pop, count in Counter(pop_pc_pd[output_col]).items())
+    ))
+
+    if hail_input:
+        pops_ht = hl.Table.from_pandas(pop_pc_pd, key=list(pop_pca_scores.key))
+        pops_ht.annotate_globals(
+            assign_pops_from_pc_params=hl.struct(
+                min_assignment_prob=min_prob
+            )
+        ).persist()
+        return pops_ht, pop_clf
+    else:
+        return pop_pc_pd, pop_clf
+
+
+def merge_stats_counters_expr(stats: hl.expr.ArrayExpression) -> hl.expr.StructExpression:
+    """
+    Merges multiple stats counters, assuming that they were computed on non-overlapping data.
+    Examples:
+    - Merge stats computed on indel and snv separately
+    - Merge stats computed on bi-allelic and multi-allelic variants separately
+    - Merge stats computed on autosomes and sex chromosomes separately
+
+    :param hl.expr.ArrayExpression stats: An array of stats counters to merge
+    :return: Merged stats Struct
+    :rtype: hl.expr.StructExpression
+    """
+
+    def add_stats(i: hl.expr.StructExpression, j: hl.expr.StructExpression) -> hl.expr.StructExpression:
+        """
+        :param hl.expr.Tuple i: accumulator: struct with mean, n and variance
+        :param hl.expr.Tuple j: new element: stats_struct -- needs to contain mean, n and stdev
+        :return: Accumulation over all elements: struct with mean, n and variance
+        :rtype: hl.expr.StructExpression
+        """
+        delta = j.mean - i.mean
+        n_tot = i.n + j.n
+        return hl.struct(
+            min=hl.min(i.min, j.min),
+            max=hl.max(i.max, j.max),
+            mean=(i.mean * i.n + j.mean * j.n)/n_tot,
+            variance=i.variance + (j.variance + (delta * delta * i.n * j.n) / n_tot),
+            n=n_tot,
+            sum=i.sum + j.sum
+        )
+
+    metrics = set(stats[0])
+    for stat_expr in stats[1:]:
+        metrics = metrics.intersection(set(stat_expr))
```

Comment:
this will drop stats that are unique to one dataset, right? maybe there should be a warning about the stats that are dropped?